platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/LongRunningAverage.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.utility;
+
+import static com.swirlds.common.formatting.StringFormattingUtils.formattedList;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * <p>
+ * Tracks a running average of the last N long values. This class does not "forget" values based on time, it
+ * only cares about the last N values regardless of when they were added relative to the clock. It does not
+ * do any special weighting, all of the last N values are weighted equally.
+ * </p>
+ *
+ * <p>
+ * By design, this class is not thread safe. If thread safety is require then the caller must synchronize.
+ * </p>
+ */
+public class LongRunningAverage {
```

Comment:
Side note:
There is a way to do this with an Accumulator, which might be simpler, but only if you want a permanent average, rather than a bounded average of <N> elements.
You could do a bounded average with an accumulator too, but that would likely be more work than really necessary.

Reply:
I think a permanent average is undesirable in this scenario. I use this running average for making an intelligent guess for the generational boundaries of a preconsensus event stream file, with the goal of choosing boundaries that give us the target file size without significant file generational overlap. If we have changing network conditions, I'd like for the code to respond quickly. In this use case, recent history is much more important than ancient history.

Also, there is no need for thread safety here. In my use case, only a single thread reads/writes this, and so there is little impetus to make changes with the goal of supporting multi-threaded operation.